docs/scaling.md

```diff
@@ -0,0 +1,119 @@
+    - Step-1: check if there is any changes in  3-resources inside scaling section: javaoptions,cpu and memory. 
+    - Step-2: ES-setting-change:  change default time from 1 min to 6 min to avoid copying of shards belonging to the data node that is going to be scaled.
+    - step-2: ES-setting-change:  change translong durability from async to sync/request basis, this is to make no data is loss.
+    - Step-3: ES-change: check if ES cluster is green state, suppose if one of the data node is down and state is yellow then do not proceed with scaling.
+    - Step-4: scale the Data node by updating the new resources in the statefull set. Here the Data node will be restarted. 
+    - Step-5: check if the POD is restarted and in running state from k8 point of view.
+    - Step-6: check if the POD is up from the ES point of view. means Data node is register with the master.
+    - Step-7: check if all shards are registered with Master, At this ES should turn in to green from yellow, now it is safe to scale next data node.
+    - Step-8: Undo the  settings done in step-2.
+ - Future Enhancements:
+    - Horizontal scaling of Data nodes: increasing the "data-node-replica" will function as expected, but if "data-node-replica" is decreased by more then 2 then the Elastic search cluster can enter in to red state and there will be data loss, this can prevented by executing similar to vertical scaling one after another without entering into red state.
+    - Picking "masternodeip" from services k8 objects instead from user.
+    - Vertical scaling improvements: periodic scaling: currently vertical scaling is triggered from user, instead it can be triggered based on time automatically.
+    - Multiple threads: Currently vertical for each elasticsearch cluster takes considerable amount of time, during this time other elastic cluster MAY not be done, this can be parallelised by running scaling operation in multiple threads.
+    - elasticsearch api's TODO: change is settings assuming the default settings.
+    - Dependency on ElasticSearch Version: It it depends on elasticsearch version as 6.3, if there is lot of changes in  api's then scaling operation will fail. 
+    - Local disk support: added changes to support local disk, but not tested with multiple nodes, it need to check that the pod will have affinity to the node, this can be enforced during the statefull set creation..
+  - Usecases:
+    - case1: scale down during start of non-peak time , and scale-up during the start of peak time every day. doing scaling operation twice in a day, this can done using a cron job without interrupting the service.
+    - case2: scale down and scale up once in a while.
+ 
+
+```
+Example Spec containing optional scaling section
+
+spec:
+  client-node-replicas: 3
+  data-node-replicas: 3
+  data-volume-size: 10Gi
+  java-options: -Xms256m -Xmx256m
+  master-node-replicas: 2
+  scaling:
+    java-options: -Xms1052m -Xmx1052m
+    masternodeip: "10.12.16.132:30002"
```

Comment:
This is the service ip of the statefuleset? Why do you need to hard code this? Shouldn't any available master be able to take over requests?

Reply:
Yes, this can be fetched from the service, Will be addressed in next commit.